src/App.jsx

```diff
@@ -14,70 +18,81 @@ export const goods = [
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const handleSelect = (good) => {
+    setSelectedGood(
+      good === selectedGood
+        ? ''
+        : good,
+    );
+  };
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {
+          !selectedGood
+            ? 'No goods selected'
+            : `${selectedGood} is selected`
+        }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+        {
+          selectedGood && (
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+              className="delete ml-3"
+              onClick={() => handleSelect('')}
+            />
+          )
+        }
+      </h1>
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+      <table className="table">
+        <tbody>
+          {
+            goods.map(good => (
+              <tr
+                data-cy="Good"
+                className={classNames({
+                  'has-background-success-light': good === selectedGood,
+                })}
+                key={good}
+              >
+                <td>
+                  <button
+                    data-cy={
+                      good === selectedGood
+                        ? 'RemoveButton'
+                        : 'AddButton'
+                    }
+                    type="button"
+                    className={classNames(
+                      'button',
+                      { 'is-info': good === selectedGood },
+                    )}
+                    onClick={() => handleSelect(good)}
+                  >
+                    {
+                      good === selectedGood
+                        ? '-'
+                        : '+'
+                    }
+                  </button>
+                </td>
```

Comment:
if you change almost everything depending on the condition, it will be better to rewrite it like this
```
{good === selectedGood 
 ? (<button></button>)
  : (<button></button>)}
```